source/module_hsolver/iterdiagcon.h

```diff
@@ -0,0 +1,51 @@
+#ifndef ITERDIAGCONTROL_H
```

Comment:
I'm sorry but this name confuses me.

Reply:
This class aims to assist Iterative diagonalization method, any better suggestion?

Reply:
> Iterative diagonalization

Following the convention of diago_iter would be better. Can we put this helper function within it?

Reply:
ok